lib/managers/window_manager.h

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <vector>
+
+#include "graphics/libs.h"
+
+typedef size_t WindowID;
```

Comment:
Consider making this definition a public member of `WindowManager`, as it has no meaning outside of the manager's responsibility sphere.

Plus, why not use
```C++
using WindowId = size_t;
```
as a more C++-y alternative to `typedef`?